test/integration/api/submissions.js

```diff
@@ -982,6 +982,32 @@ describe('api: /forms/:id/submissions', () => {
                 done();
               })))))));
 
+    it('should filter attachments by the query', testService((service) =>
+      service.login('alice', (asAlice) =>
+        service.login('bob', (asBob) =>
+          asAlice.post('/v1/projects/1/forms?publish=true')
+            .set('Content-Type', 'application/xml')
+            .send(testData.forms.binaryType)
+            .expect(200)
+            .then(() => asAlice.post('/v1/projects/1/submission')
+              .set('X-OpenRosa-Version', '1.0')
+              .attach('xml_submission_file', Buffer.from(testData.instances.binaryType.one), { filename: 'data.xml' })
+              .attach('my_file1.mp4', Buffer.from('this is test file one'), { filename: 'my_file1.mp4' })
+              .expect(201)
+              .then(() => asBob.post('/v1/projects/1/submission')
+                .set('X-OpenRosa-Version', '1.0')
+                .attach('xml_submission_file', Buffer.from(testData.instances.binaryType.two), { filename: 'data.xml' })
+                .attach('here_is_file2.jpg', Buffer.from('this is test file two'), { filename: 'here_is_file2.jpg' })
+                .expect(201))
+              .then(() => new Promise((done) =>
+                zipStreamToFiles(asAlice.get('/v1/projects/1/forms/binaryType/submissions.csv.zip?$filter=__system/submitterId eq 5'), (result) => {
+                  result.filenames.should.containDeep([
+                    'binaryType.csv',
+                    'media/my_file1.mp4'
+                  ]);
```

Comment:
Should this test that `here_is_file2.jpg` isn't in the .zip?

Reply:
yeah probably